Riot/Modules/Home/AllChats/AllChatsViewController.swift

```diff
@@ -500,22 +509,40 @@ class AllChatsViewController: HomeViewController {
             self?.updateToolbar(with: menu)
         }))
         updateEmptyView()
+        updateBadgeButton()
     }
     
     private func updateRightNavigationItem(with menu: UIMenu) {
         self.navigationItem.rightBarButtonItem = UIBarButtonItem(image: UIImage(systemName: "ellipsis.circle"), menu: menu)
     }
     
+    private lazy var spacesButton: BadgedBarButtonItem = {
+        let innerButton = UIButton(type: .system)
+        innerButton.accessibilityLabel = VectorL10n.spaceSelectorTitle
+        innerButton.addTarget(self, action: #selector(self.showSpaceSelectorAction(sender:)), for: .touchUpInside)
+        innerButton.setImage(Asset.Images.allChatsSpacesIcon.image, for: .normal)
+        return BadgedBarButtonItem(withBaseButton: innerButton, theme: theme)
+    }()
+    
+    @objc private func updateBadgeButton() {
+        guard isViewLoaded else {
+            return
+        }
+        spacesButton.badgeText = session.map {
+            "\($0.spaceService.rootSpacesNotificationCount)"
+        }
+        spacesButton.badgeBackgroundColor = session.map {
+            $0.spaceService.rootSpacesHaveHighlightNotification ? theme.noticeColor : theme.noticeSecondaryColor
+        } ?? .clear
```

Comment:
Mmmh, the logic looks different.
Here we show the count of all unread messages including the ones in **current space**.
Invites to spaces are ignored unfortunately! 😅
@Johennes do you think it makes sense to cover also invites under this PR (changing the ACs)?
Not having them covered looks like a UX bug to me.

Reply:
> Here we show the count of all unread messages including the ones in current space.

My bad I hadn't noticed that 🙈

Reply:
One thought looking at https://github.com/vector-im/element-ios/pull/7088/commits/f1ce202f808612b0f416820ff573c8919b83f845. Moving the `notificationState(forAllSpacesExcept: nil)` up to this method would mean you could read `allCount` and `allHighlightCount` without having to iterate through the spaces twice.

Reply:
Including space invites in the highlight count seems sensible to me as I assume you otherwise might miss them just like normal messages in another space than the one you're in?

Reply:
Yeah @Johennes, you wouldn't notice invites.
the tricky thing is that invites aren't countable (at least in the design we have currently for spaces).
They are "exclamation mark" badges. :-/ 

![Simulator Screen Shot - iPhone 14 Pro - 2022-11-18 at 16 38 52](https://user-images.githubusercontent.com/19324622/202743128-ddf4b056-0bef-41ed-8ed9-313bad1ad468.png)

Reply:
Oh interesting. Maybe it still makes sense to sum that up? It's still an unread message of sorts so I suppose it should count as a 1.

Reply:
Or alternatively we could only use it to drive the color of the spaces icon badge?

Reply:
If we can go this way probably we would need to implement both.
If you rely just on the color you could find yourself in a situation where you don't have unread messages but just invites and the badge wouldn't show. :-/

Reply:
> If you rely just on the color you could find yourself in a situation where you don't have unread messages but just invites and the badge wouldn't show. :-/

Could we just use the red exclamation mark on the spaces icon in that case?

> Side note to do with as you wish: I was referring to invites as "an invite to a DM which only show up in the All Chats screen".

Oh, I see! That's this case then:

![simulator_screenshot_F08BEE4B-2C68-47EB-A56D-8695A6C6A65B](https://user-images.githubusercontent.com/1137962/202752962-a925413e-c65c-433a-b4c3-6ba12126a69a.png)

Well, I suppose we should be covering both. This one seems more straightforward though because the invite can simply count as a red badge of 1 in the all spaces space?